lib/cartopy/tests/mpl/test_mpl_integration.py

```diff
@@ -654,6 +654,21 @@ def test_pcolormesh_single_column_wrap():
     return fig
 
 
+def test_pcolormesh_wrap_gouraud_shading_failing_mask_creation():
+    x_range = np.linspace(-180, 180, 50)
+    y_range = np.linspace(90, -90, 50)
+    x, y = np.meshgrid(x_range, y_range)
+    data = ((np.sin(np.deg2rad(x))) / 10. + np.exp(np.cos(np.deg2rad(y))))
+
+    fig = plt.figure(figsize=(10, 6))
+    ax = fig.add_subplot(1, 1, 1, projection=ccrs.Mercator())
+    try:
+        ax.pcolormesh(x, y, data, transform=ccrs.PlateCarree(), shading='gouraud')
+    except ValueError as e:
+        if str(e) == 'negative dimensions are not allowed':
+            pytest.fail("Unexpected ValueError('negative dimensions are not allowed')")
```

Comment:
pytest will fail for us, so no need to explicitly catch and fail.
```suggestion
    ax.pcolormesh(x, y, data, transform=ccrs.PlateCarree(), shading='gouraud')
```

Reply:
Yes, done. THX.